### PR TITLE
UIU-2975 - Link externally hosted image as user profile picture

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
[UIU-2975](https://folio-org.atlassian.net/browse/UIU-2975) - Profile Pictures Functionality - Link externally hosted image

## Approach
Added a hidden input field to register to the form and on user action, the value of the hidden input will be updated.

## Note
**19/29 changes files have fixes for older linting issues.**

## Screencast

https://github.com/folio-org/ui-users/assets/104053200/7cfd2002-dd0c-47f1-b5ad-c4d1ddf78e80

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by manvendra-s-rathore — APPROVED** [review]

**Comment by Terala-Priyanka** [comment]
@folio-org/fe-tl-reviewers 
Please review my PR.

File: src/components/EditSections/EditUserInfo/components/ProfilePicture/ProfilePicture.js [hunk 1]
```diff
@@ -12,33 +12,47 @@ import {
...
-        id="profile-picture"
         className={css.profilePlaceholder}
         alt={intl.formatMessage({ id: 'ui-users.information.profilePicture' })}
         src={imgSrc}
       />
     );
   };
 
+  const toggleExternalLinkModal = () => {
+    setExternalLinkModalOpen(prev => !prev);
```
Comment (OleksandrHladchenko1) [inline src/components/EditSections/EditUserInfo/components/ProfilePicture/ProfilePicture.js:46]: Let's use more meaningful values (e.g. open, isOpen)

File: src/components/EditSections/EditUserInfo/components/ProfilePicture/ProfilePicture.test.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { render, screen } from '@folio/jest-config-stripes/testing-library/react';
+import { render, screen, fireEvent } from '@folio/jest-config-stripes/testing-library/react';
```
Comment (OleksandrHladchenko1) [inline src/components/EditSections/EditUserInfo/components/ProfilePicture/ProfilePicture.test.js:1]:
```suggestion
import {
  render,
  screen,
  fireEvent,
} from '@folio/jest-config-stripes/testing-library/react';
```

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { screen, render, fireEvent } from '@folio/jest-config-stripes/testing-library/react';
```
Comment (OleksandrHladchenko1) [inline src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.test.js:1]:
```suggestion
import {
  screen,
  render,
  fireEvent,
} from '@folio/jest-config-stripes/testing-library/react';
```

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+describe('ExternalLinkModal', () => {
+  const props = {
+    onOpen: jest.fn(),
+    onClose: jest.fn(),
+    onSave: jest.fn(),
+    profilePictureLink: 'profilePictureLink',
+  };
+  beforeEach(() => {
+    renderExternalLinkModal(props);
+  });
```
Comment (OleksandrHladchenko1) [inline src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.test.js:30]: Add spaces between tests and logical components (beforeEach, props), please

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useRef, useState } from 'react';
```
Comment (OleksandrHladchenko1) [inline src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js:1]:
```suggestion
import React, {
  useEffect,
  useRef,
  useState,
} from 'react';
```

File: src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import {
+  Button,
+  Modal,
+  ModalFooter,
+  TextField,
+} from '@folio/stripes/components';
+import { FormattedMessage } from 'react-intl';
+import { isAValidURL } from '../../../../util';
+
+const ExternalLinkModal = ({ open, onClose, onSave, profilePictureLink }) => {
```
Comment (OleksandrHladchenko1) [inline src/components/EditSections/EditUserInfo/components/ExternalLinkModal/ExternalLinkModal.js:12]: Don't put more than two values in one row, please.

File: src/components/EditSections/EditUserInfo/components/ProfilePicture/ProfilePicture.js (hunk 1 above)
Comment (Terala-Priyanka) [reply src/components/EditSections/EditUserInfo/components/ProfilePicture/ProfilePicture.js:46]: As discussed, I am going with this notation, as it does the toggle job irrespective of the status of the modal - open or close.

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Pull request merged**